packages/components/src/VirtualizedList/ListGrid/ListGrid.component.js

```diff
@@ -2,58 +2,58 @@ import React, { PropTypes } from 'react';
 import { List as VirtualizedList } from 'react-virtualized';
 import getRowSelectionRenderer from '../RowSelection';
 import NoRows from '../NoRows';
-import { getRowData } from '../utils/gridrow';
 
 import theme from './ListGrid.scss';
 
-function getRowDataFromParent({ parent, index }) {
-	return getRowData(parent, index);
-}
-
 /**
  * List renderer that accepts a custom row renderer.
  * The row renderer will create a row element for each collection item.
  */
-export default function ListGrid(props) {
-	const {
-		children,
-		collection,
-		id,
-		height,
-		isSelected,
-		rowHeight,
-		rowRenderer,
-		selectionToggle,
-		width,
-	} = props;
-
-	let enhancedRowRenderer = rowRenderer;
-	if (selectionToggle) {
-		enhancedRowRenderer = getRowSelectionRenderer(
+class ListGrid extends React.Component {
+	render() {
+		const {
+			children,
+			collection,
+			id,
+			height,
+			isSelected,
+			rowHeight,
 			rowRenderer,
-			{
-				isSelected,
-				getRowData: getRowDataFromParent,
-			}
+			selectionToggle,
+			width,
+		} = this.props;
+
+		this.collection = collection;
```

Comment:
I can't. If I do that, when rowGetter will be call after a change of "collection", it will be a reference to the old props and i 'll still have the same bug.

Every time, after that collection change, we do a reference to the old collection.